File: 0409-longest-palindrome/0409-longest-palindrome.dart
class Solution {
  int longestPalindrome(String s) {
    dynamic counter = {};
    var mx = 0;
    dynamic answer = 0;
    for (var i = 0; i < s.length; i++) {
        if (counter.containsKey(s[i])) {
            counter[s[i]] += 1;
        } else {
            counter[s[i]] = 1;
        }
    }
    print(counter.values);
    for (var n in counter.values) {
        if (n % 2 == 1) {
            if (mx <= n) {
                if (mx != 0) {
                    answer += mx - 1;
                }
                mx = n;
            } else {
                answer += n - 1;
            }
        } else {
            answer += n;
        }
    }
    answer += mx;
    return answer;
  }
}

File: 0409-longest-palindrome/0409-longest-palindrome.py
from collections import Counter
class Solution:
    def longestPalindrome(self, s: str) -> int:
        c_s = Counter(s)
        answer = 0
        flag = False
        for a, b in c_s.most_common():
            if b % 2:
                if b > 1:
                    answer += b - 1
                    flag = True
                else:
                    flag = True
            else:
                answer += b
        else:
            if flag:
                answer += 1
        
        return answer